### DISCUSSION TITLE
QueryParams and Headers are both multidicts but behave differently

### DISCUSSION THREAD

**Discussion by rafalkrupinski** [encode/starlette#2667] [Potential Issue]
...and I don't mean that one is case sensitive and the other isn't

When receiving multi-valued parameter (httpx client):
```py
server.get('http://example.com', params=httpx.QueryParams([
        ('param', 'value1'),
        ('param', 'value2')
    ]))
```
`request.query_params.keys()` returns `param` once, and I can get each value from `request.query_params.getlist('param')`

When receiving multi-valued header:
```py
client_server.get('http://example.com', headers=httpx.Headers([
        ('param', 'value1'),
        ('param', 'value2'),
    ]))
```
`request.headers.keys()` returns `param` twice.

Not a critical issue, but confusing. If they share API, they should share behaviour.

**Comment by adriangb** [comment]
I do think we could unify the APIs to return a single key from `.keys()`.
